### PR TITLE
refactor(YouTube - Spoof streaming data): Remove `Force iOS AVC` setting

### PR DESCRIPTION
**Pull request by YT-Advanced**
We already have `Disable VP9 Codec` patch, which did the same thing. Therefore, keep `Force AVC` settings can cause some confusion for users.

Moreover, `DeviceHardwareSupport` is not needed, as @kitadai31 mentioned in https://github.com/ReVanced/revanced-patches/pull/4064#issuecomment-2525542497

### CODE REVIEW

File: patches/src/main/kotlin/app/revanced/patches/youtube/video/playback/VideoPlaybackPatch.kt [hunk 1]
```diff
@@ -315,7 +315,7 @@ val videoPlaybackPatch = bytecodePatch(
 
         // region patch for disable VP9 codec
 
-        vp9CapabilityFingerprint.methodOrThrow().apply {
+        vp9CapabilityFingerprint.methodOrThrow(hdrCapabilityFingerprint).apply {
```
Comment (inotia00) [inline patches/src/main/kotlin/app/revanced/patches/youtube/video/playback/VideoPlaybackPatch.kt]: This is an unnecessary change

Comment (YT-Advanced) [reply patches/src/main/kotlin/app/revanced/patches/youtube/video/playback/VideoPlaybackPatch.kt]: vp9CapabilityFingerprint and hdrCapabilityFingerprint are in the same class, so do it will speed up the resolve time

**Comment by inotia00** [comment]
The `Disable VP9 codec` setting and the `Force iOS AVC` setting seem to do similar things, but they are a bit different

1. Turn on `Spoof streaming data`, turn on `Disable VP9 codec`
: Some videos have a max resolution of 360p in AVC codec

2. Turn on `Spoof streaming data`, turn on `Force iOS AVC`
: Some videos have a max resolution of 1080p in AVC codec

As the ReVanced Team did, the `Force iOS AVC` setting will be back
